src/imitation/algorithms/preference_comparisons.py

```diff
@@ -1355,6 +1456,117 @@ def _gather_preference(self, query_id: str) -> float:
         return answered_query["label"]
 
 
+class ZooniverseGatherer(PrefCollectGatherer):
+    """Gathers preferences from Zooniverse interface."""
+
+    def __init__(
+        self,
+        pref_collect_address: str,
+        zoo_project_id: int,
+        zoo_workflow_id: int,
+        linked_subject_set_id: int,
+        wait_for_user: bool = True,
+        rng: Optional[np.random.Generator] = None,
+        custom_logger: Optional[imit_logger.HierarchicalLogger] = None,
+        querent_kwargs: Optional[Mapping] = None
+    ) -> None:
+        """Initializes the preference gatherer.
+
+        Args:
+            pref_collect_address: Network address to PrefCollect instance.
+            wait_for_user: Waits for user to input their preferences.
+            rng: random number generator, if applicable.
+            custom_logger: Where to log to; if None (default), creates a new logger.
+        """
+        video_output_dir = querent_kwargs["video_output_dir"]
+        super().__init__(pref_collect_address, querent_kwargs={"video_output_dir": querent_kwargs["video_output_dir"]})
+        self.querent = ZooniverseQuerent(
+            pref_collect_address,
+            **querent_kwargs
+        )
+        
+        self.zoo_project_id = zoo_project_id
+        self.zoo_workflow_id = zoo_workflow_id
+        self.linked_subject_set_id = linked_subject_set_id
+        
+        # Find workflow
+        self.workflow = Workflow.find(self.zoo_workflow_id)
+        
+        # Define annotation to label map
+        self.annotation_to_label = {
+            0: 1,
+            1: 0,
+            2: .5,
+            3: -1,
+            None: None
+        }
+        
+        self.last_id = 0
+        self.query_to_subject = {}
+        self.subject_to_annotations = {}
+        
+    def _process_zoo_classifications(self):
+        
+        # Access classifications from the last_id
+        classifications = Classification.where(
+            last_id=self.last_id,
+            scope='project',
+            project_id=self.zoo_project_id,
+            workflow_id=self.zoo_workflow_id
+        )
+        
+        # get linked subjects and their statuses
```

Comment:
instead of this comment, you could encapsulate this in a small method whose name could be `_get_linked_subject_statuses()` -(returns)-> `linked_subject_statuses`